addons/sale_timesheet/models/project.py

```diff
@@ -382,18 +380,9 @@ def _compute_analytic_account_active(self):
 
     @api.depends('sale_line_id', 'project_id', 'allow_billable', 'commercial_partner_id')
     def _compute_sale_order_id(self):
```

Comment:
Shouldn't most of the dependencies be dropped now that only `allow_billable` is directly used by this method?

Reply:
Don't we need to write explicitly again the dependencies of the parent method if we need them?

Reply:
Nope, each override only needs to specify its own needs otherwise it would be amodular / unusable for reasons of not knowing what other overrides there are, and thus which the next method is e.g. lets say that you have three methods

    @depends('a', 'b')
    def _compute_thing(self):
        ...

    # override 1
    @depends('a', 'b', 'c')
    def _compute_thing(self):
        ...

    # override 2
    @depends('a', 'b', 'd')
    def _compute_thing(self):
        ...

If "override 1" and "override 2" are in independent modules, especially developed by different people entirely (e.g. from the appstore) there is no way for the developers of those to reconcile the computes, there would need to be a link module for every possible combination of modules (at least sharing overrides) which is infeasible.

Therefore a compute override should only specify its own dependencies, the ORM ought take care of combining the dependencies of the base & overrides when it compiles the "live" model.

Note: that doesn't mean you should leave off dependencies which you "know" the base method has, the base method can be udpated at any point and its set of dependencies modified arbitrarily. So an override should *only* specify its own dependencies, but it should specify all of them (usually).